core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -242,7 +242,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (forcedRefresh) {
             builder.field("forced_refresh", forcedRefresh);
         }
-        shardInfo.toXContent(builder, params);
+        builder.field("_shards", shardInfo);
```

Comment:
I did it so that ShardInfo.toXContent() starts with a startObject() and not a field name, but it is not a requirement. It's just easier to test since there's no need to wrap it into an anonymous object.